### PR TITLE
Add UMNN

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Add UMNN v1.71 by Antoine Wehenkel (@AWehenkel).
   - c.f. https://github.com/AWehenkel/UMNN
* c.f. https://github.com/bayesiains/nflows/issues/82

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge/help-python, ready for review!

File: recipes/umnn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{% set name = "umnn" %}
+{% set version = "1.71" %}
+{% set python_min = "3.7" %}
```
Comment (ocefpaf) [inline recipes/umnn/meta.yaml:3]: Be aware that conda-forge min supported Python is 3.9.

Comment (matthewfeickert) [reply recipes/umnn/meta.yaml:3]: Thanks for the fast merge, @ocefpaf!

Indeed, that's why I set the `python_min` here to try to be as close to PyPI as could be supported. My understanding (please correct me if I'm wrong here) is that the current global pins (so supported versions) are [Python 3.9 to 3.12](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L826-L841) and that these are the default supported versions, but, if you wanted to go outside those supported defauts that's still okay for `noarch: python` packages as long as you don't expect any support there. Is that correct?

Comment (ocefpaf) [reply recipes/umnn/meta.yaml:3]: It is correct. The warning is mostly "if this breaks, it is unlikely we'll supported it." (It should work though.)

Comment (matthewfeickert) [reply recipes/umnn/meta.yaml:3]: Great and thanks very much for the info! I really appreciate the work that you and the other core team members do to keep everything running so smoothly.

**Pull request merged**